File: nodejs-wdio/tests/pages/webinars.page.ts
import Page from "./page";
import SignInPage from "./signIn.page";
import AccountMenuComponent from "./components/accountMenu.comp";
import isSignedIn from "../helpers/signIn.helper";

const SCREEN_SELECTOR = "div.title"; // Nice seeing you again

class WebinarsPage extends Page {
  constructor() {
    super(SCREEN_SELECTOR);
  }

  public isSignedIn(signIn = true): boolean {
    return isSignedIn(signIn);
  }

  public signOut(): typeof SignInPage {
    AccountMenuComponent.signOut();
    return SignInPage;
  }
}

export default new WebinarsPage();

File: ruby-capybara/tests/pages/webinars_page.rb
require_relative '../pages/base_page'
require_relative '../pages/sign_in_page'
require_relative '../pages/components/account_menu_comp'

# Page Webinars
class WebinarsPage < BasePage
  SCREEN_SELECTOR = 'div.title'
  def initialize()
    super(SCREEN_SELECTOR)
    @account_menu = AccountMenuComponent.new
  end

  def sign_out
    @account_menu.sign_out
  end
end
